include/nodes/internal/ConnectionStyle.hpp

```diff
@@ -43,18 +34,35 @@ class NODE_EDITOR_PUBLIC ConnectionStyle : public Style
 
   bool useDataDefinedColors() const;
 
+  void setConstructionColor(QColor);
+  void setNormalColor(QColor);
+  void setSelectedColor(QColor);
+  void setSelectedHaloColor(QColor);
+  void setHoveredColor(QColor);
```

Comment:
Should we go further with the Qt-kung-fu and describe all the members in the class with [`Q_PROPERTY`](http://doc.qt.io/qt-5/properties.html) ?

As I understand, something like this 

    Q_PROPERTY(QColor color MEMBER m_color NOTIFY colorChanged)

would define a private member and generate setters and getters.

Reply:
Not a bad idea. E.g. when we implement a QML frontend, that could come in handy. IIUC, `Q_PROPERTY` only has meaning inside something that inherits from `QObject`, which the `FooStyle`s currently don't. `QObject`s can't be copied, so there are some options:

* Pass around `std::unique_ptr<FooStyle>` instead of `FooStyle` directly.
* Make `FooStyle` take a `QObject* parent = Q_NULLPTR` in its constructor.
* Maybe this could work?: Write explicit copy constructors for `FooStyle`s